Add support for API version 2019-03-14

remi-stripe opened the pull request:
The changes in the library itself look minor. But this API version changes the behavior of Subscriptions entirely under the hood. If the charge fails, you still get a Subscription back instead of an exception and it's on you to check the Subscription status.

This is now fully ready to be reviewed. It also changes the `apiversion` internal-only constant to a public one.

r? @brandur-stripe 
cc @stripe/api-libraries 

Related to https://github.com/stripe/stripe-dotnet/pull/1552

brandur-stripe commented:
> The changes in the library itself look minor. But this API version changes the behavior of Subscriptions entirely under the hood. If the charge fails, you still get a Subscription back instead of an exception and it's on you to check the Subscription status.

Oh, very good to know! Thanks.

LGTM.

remi-stripe commented:
@brandur-stripe I ended up finding a bug in the library which I fixed in https://github.com/stripe/stripe-go/pull/814. After that one is merged, we can properly force the new stripe-mock flag to test the API version in the library is the correct one.

Because of this, `apiVersion` now has to be exported so I renamed it. 

PTAL

brandur-stripe commented:
> Because of this, `apiVersion` now has to be exported so I renamed it. 

IMO, this is okay. It may be useful to print informational logs or something in client applications anyway. As long as it stays as a `const`, it can't be abused (by people overwriting it to a new value, etc.).

brandur-stripe commented:
Released as 58.0.0.
